src/autowiring/test/ContextCreatorTest.cpp

```diff
@@ -200,3 +200,35 @@ TEST_F(ContextCreatorTest, VoidKeyType) {
   EXPECT_EQ(0UL, vc->GetSize()) << "A void context creator was not correctly updated when its dependent context went out of scope";
   EXPECT_EQ(2UL, vc->m_totalDestroyed) << "The void creator did not receive the expected number of NotifyContextDestroyed calls";
 }
+
+struct mySigil {};
+
+class BackReferencingClass:
+  public CoreRunnable
+{
+public:
+  bool OnStart(void) override {
+    AutoCurrentContext ctxt;
+    m_parentContext = ctxt->GetParentContext();
```

Comment:
Why the parent context and not the current context?  Can you add a comment here to describe your reasoning?